tailormap-components/projects/core/src/lib/user-interface/attributelist/layer.service.ts

```diff
@@ -24,11 +23,43 @@ export class LayerService {
               private highlightService: HighlightService) {
     // Install the layerVisibilityChanged handler.
     this.tailorMapService.layerVisibilityChanged$.subscribe(value => {
-      this.loadLayers();
+      // layerVisibilityChanged visible to true occurs too often (also if layer is already visible)
+      // console.log ('LayerService visi changed value: ' + value.visible);
+      if (value.visible) {
+        if (!this.isLayerIdInLayers(value.layer.id)) {
+          this.addLayer(value.layer.id);
+        }
+      } else {
+        if (this.isLayerIdInLayers(value.layer.id)) {
+          this.removeLayer(value.layer.id);
+        }
+      }
+      this.layersSubject.next(this.layers);
     });
     this.loadLayers();
   }
 
+  /**
+   * Returns part from full layer name before ":". Converts to lowercase too.
+   */
+  private static sanitizeLayername(layerName: string): string {
+    const index = layerName.indexOf(':');
+    if (index !== -1) {
+      layerName = layerName.substring(index + 1);
+    }
+    return layerName.toLowerCase();
+  }
```

Comment:
Gebruik hiervoor de LayerUtils.sanitizeLayername()

Reply:
Hm, deze zat er al in, maar die heb ik static gemaakt.

Reply:
Aangepast en ook LayerUtils: sanitzeLayername -> sanitizeLayername